### PR TITLE
[mlir][gpu] Improve `gpu-lower-to-nvvm-pipeline` Documentation

### PR DESCRIPTION
**Pull request by grypp**
This PR improves the documentation for the `gpu-lower-to-nvvm-pipeline` (as it was remaning item for #75775)

- Changes pipeline `gpu-lower-to-nvvm` -> `gpu-lower-to-nvvm-pipeline`
- Adds a section in GPU Dialect in website. It clarifies the pipeline's functionality in lowering primary dialects to NVVM targets.

### CODE REVIEW

**Comment by grypp** [comment]
I continued to provide support for 'linalg' with 'ConvertLinalgToLoops.' I mentioned [here](https://github.com/grypp/llvm-project/blob/nvgpu-compiler/mlir/docs/Dialects/GPU.md?plain=1#L67), the pipeline requires explicitly parallel IR and doesn't do parallelism to any of the operations especially for `linalg`. Does this explanation align with our goals?

**Comment by joker-eph** [comment]
> The pipeline requires explicitly parallel IR and doesn't do parallelism to any of the operations especially for linalg. Does this explanation align with our goals?

I would think that "gpu" is already parallel code (the pipeline is "gpu-to-nvvm" here): so it seems to me that handling of linalg should be done outside of this pipeline  and this pipeline invoked on the results.

File: mlir/docs/Dialects/GPU.md [hunk 1]
```diff
@@ -60,6 +60,38 @@ mlir-translate example-nvvm.mlir        \
   -o example.ll
 ```
 
+### Default NVVM Compilation Pipeline: gpu-lower-to-nvvm-pipeline
+
+The `gpu-lower-to-nvvm-pipeline` compilation pipeline serves as the default way for NVVM target compilation within MLIR. This pipeline operates by lowering primary dialects (arith, linalg, memref, scf, vector, gpu, and nvgpu) to NVVM target. It begins by lowering GPU code region(s) to the specified NVVM compilation target and subsequently handles the host code.
+
+This pipeline specifically requires explicitly parallel IR and doesn't do GPU parallelization. For example, it supports linalg but it gets translated into sequential loops. To incorporate parallelism, necessary transformations must be applied before utilizing this pipeline.
+
+It's designed to provide a generic solution for NVVM targets, generating NVVM and LLVM dialect code compatible with `mlir-cpu-runner` or execution engine.
```
Comment (joker-eph) [inline mlir/docs/Dialects/GPU.md:69]: Can you wrap this? `mdformat -wrap 80` should do.

**Comment by grypp** [comment]
> > The pipeline requires explicitly parallel IR and doesn't do parallelism to any of the operations especially for linalg. Does this explanation align with our goals?
> 
> I would think that "gpu" is already parallel code (the pipeline is "gpu-to-nvvm" here): so it seems to me that handling of linalg should be done outside of this pipeline and this pipeline invoked on the results.

I see yor point, it makes sense. I removed the lowering of linalg. In integration tests I added `convert-linalg-to-loops` before `gpu-to-nvvm-pipeline`.

**Review by joker-eph — APPROVED** [review]
Thanks!

**Pull request merged**